File: leetcode_problems/typescript/src/lc2073_time_needed_to_buy_tickets.ts
function timeRequiredToBuy(tickets: number[], k: number): number {
  let res: number = 0;
  let target: number = tickets[k];
  for (let i = 0; i < tickets.length; i++) {
    if (i <= k) {
      res += Math.min(target, tickets[i]);
    } else {
      res += Math.min(target - 1, tickets[i]);
    }
  }
  return res;
}

console.log(timeRequiredToBuy([5, 1, 1, 1], 0));

File: leetcode_problems/cpp/lc2073_time_needed_to_buy_tickets.cpp
#include <vector>
#include <iostream>
using std::vector;

class Solution
{
public:
    int timeRequiredToBuy(vector<int> &tickets, int k)
    {
        int res = 0;
        int target = tickets[k];
        for (int i = 0; i < tickets.size(); i++)
        {
            if (i <= k)
            {
                res += std::min(target, tickets[i]);
            }
            else
            {
                res += std::min(target - 1, tickets[i]);
            }
        }
        return res;
    }
};

int main()
{
    vector<int> tickets = {5, 1, 1, 1};
    int k = 0;
    Solution sol;
    int res = sol.timeRequiredToBuy(tickets, k);
    std::cout << res << std::endl;
    return 0;
}